k4FWCore/src/PodioDataSvc.cpp

```diff
@@ -109,14 +126,20 @@ StatusCode PodioDataSvc::i_setRoot(std::string root_path, DataObject* pRootObj)
 }
 
 void PodioDataSvc::endOfRead() {
+  m_eventNum++;
+
+  if (!m_bounds_check_needed) {
+    return;
+  }
+
   StatusCode sc;
-  if (m_eventMax != -1) {
-    if (m_eventNum++ >= m_eventMax - 1) {  // we start counting at 0 thus the -1.
-      info() << "Reached end of file with event " << m_eventMax << endmsg;
-      IEventProcessor* eventProcessor;
-      sc = service("ApplicationMgr", eventProcessor);
-      sc = eventProcessor->stopRun();
-    }
+  // m_eventNum points to the next event already so we check if its available
+  // (m_eventNum starts counting from 0, so <)
+  if (!(m_eventNum < m_availableEventMax)) {
```

Comment:
@jmcarcell how do you like this?

Reply:
I don't think they are equivalents, if `m_availableEventMax == -1` then the new version can trigger while the old one wouldn't. The check is fine, I don't think we need to change it, it's just the commenting and naming

Reply:
true :/ I will roll this back